### PR TITLE
Code commit for RT-3.1: Policy based VRF selection

### LINKED ISSUE #149: RT-3.1: [Protocol, DSCP]-based VRF Selection

### PR DESCRIPTION
**Pull request by smulky**
Creating new request due to some merge issue with https://github.com/openconfig/featureprofiles/pull/228 
Resolves #149

### CODE REVIEW

**Comment by dplore** [comment]
@michaelore for go-reviewer

**Comment by bstoll** [comment]
I spent some time working on this at https://github.com/bstoll/featureprofiles/tree/base_vrf_selection.  I noticed the forwarding policy being configured does not align with the test plan, and seems to include Discard operations for specific prefixes.  I am not sure if this is all necessary; it looks like it should be possible to express the operations almost exactly as they are defined in the test plan in 1:1 policies to each test case.  However, I have run into some issues that I have not been able to sufficiently test these changes to feel confident that my work is not avoiding some issue you ran into.

If you have a chance @smulky , I would like to see if the cases particularly in https://github.com/bstoll/featureprofiles/blob/578aa495d436e7e4ad411d0b895917337bdbf9c0/feature/experimental/policy/policy_vrf_selection/ate_tests/base_vrf_selection/base_vrf_selection_test.go#L179-L204 make sense to you.

File: feature/experimental/policy/policy_vrf_selection/ate_tests/base_vrf_selection/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# RT-3.1: Policy based VRF selection
```
[the remark this answers is not in the document]
Comment (azubaidhar) [reply feature/experimental/policy/policy_vrf_selection/ate_tests/base_vrf_selection/README.md:1]: Understood. will commit the recent change shortly.

File: feature/experimental/policy/policy_vrf_selection/ate_tests/base_vrf_selection/base_vrf_selection_test.go [hunk 2]
```diff
@@ -0,0 +1,487 @@
...
+
+// Apply forwarding policy on the interface
+func applyForwardingPolicy(t *testing.T, ate *ondatra.ATEDevice, ingressPort string, matchType string) {
+
+	t.Logf("*** Applying forwarding policy %v on interface %v ... ", matchType, ingressPort)
+
+	d := &telemetry.Device{}
+	dut := ondatra.DUT(t, "dut")
+
+	intf := d.GetOrCreateNetworkInstance("DEFAULT").GetOrCreatePolicyForwarding().GetOrCreateInterface("et-0/0/1.1")
```
[the remark this answers is not in the document]
Comment (azubaidhar) [reply feature/experimental/policy/policy_vrf_selection/ate_tests/base_vrf_selection/base_vrf_selection_test.go:285]: Understood. will commit the recent change shortly.

File: feature/experimental/policy/policy_vrf_selection/ate_tests/base_vrf_selection/base_vrf_selection_test.go [hunk 3]
```diff
@@ -0,0 +1,487 @@
...
+	intf.ApplyForwardingPolicy = ygot.String(matchType)
+	intf.GetOrCreateInterfaceRef().Interface = ygot.String(ingressPort)
+	intf.GetOrCreateInterfaceRef().Subinterface = ygot.Uint32(1)
+
+	intf2 := d.GetOrCreateNetworkInstance("DEFAULT").GetOrCreatePolicyForwarding().GetOrCreateInterface("et-0/0/1.2")
```
[the remark this answers is not in the document]
Comment (azubaidhar) [reply feature/experimental/policy/policy_vrf_selection/ate_tests/base_vrf_selection/base_vrf_selection_test.go:290]: Understood. will commit the recent change shortly.

**Comment by azubaidhar** [comment]
Thanks.

Will review https://github.com/bstoll/featureprofiles/commit/578aa495d436e7e4ad411d0b895917337bdbf9c0

File: feature/experimental/policy/policy_vrf_selection/ate_tests/base_vrf_selection/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (azubaidhar) [reply feature/experimental/policy/policy_vrf_selection/ate_tests/base_vrf_selection/README.md:1]: Fixed in https://github.com/openconfig/featureprofiles/pull/712

File: feature/experimental/policy/policy_vrf_selection/ate_tests/base_vrf_selection/base_vrf_selection_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (azubaidhar) [reply feature/experimental/policy/policy_vrf_selection/ate_tests/base_vrf_selection/base_vrf_selection_test.go:285]: Fixed in https://github.com/openconfig/featureprofiles/pull/712

File: feature/experimental/policy/policy_vrf_selection/ate_tests/base_vrf_selection/base_vrf_selection_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (azubaidhar) [reply feature/experimental/policy/policy_vrf_selection/ate_tests/base_vrf_selection/base_vrf_selection_test.go:290]: Fixed in https://github.com/openconfig/featureprofiles/pull/712

**Comment by azubaidhar** [comment]
Latest changes done in https://github.com/openconfig/featureprofiles/pull/712

**Pull request closed without merging**